cmd/check/main.go

```diff
@@ -35,6 +42,68 @@ func main() {
 		return
 	}
 
+	if req.Source.Ecr {
```

Comment:
This seems to be common between `in`, `out`, and `check` - can it be pulled out into a method on `req.Source` perhaps? Maybe `err := req.Source.ConfigureViaECR()`?